### PR TITLE
Feature/store user id in session

### PR DESCRIPTION
**Pull request by lewiscj97**
- Stores the userid when the user signs in or signs up
- This can be used when creating spaces, and also for viewing spaces by the userid

### CODE REVIEW

File: spec/lib/user_spec.rb [hunk 1]
```diff
@@ -43,4 +43,30 @@
...
+      user = User.find(id)
+      
+      expect(user.id).to eq id
+      expect(user.username).to eq 'Foo'
+      expect(user.email).to eq 'foo@bar.com'
+      expect(user.password).to eq 'protected'
+    end
+  end
+
+  describe "#get_user_id" do
```
Comment (michelle-watkins) [inline spec/lib/user_spec.rb:61]: Single quotes here for `'#get_user_id'`?

File: app.rb [hunk 2]
```diff
@@ -19,15 +20,18 @@ class MakersBnb < Sinatra::Base
...
 
   post '/sign-up' do
-    User.create(params['username'], params['email'], params['password'])
+    user = User.create(params['username'], params['email'], params['password'])
+    session[:user_id] = user.id
     redirect('/sign-up-completed')
   end
 
   get '/sign-up-completed' do
+    p session[:user_id]
```
Comment (michelle-watkins) [inline app.rb]: Can you explain why you `p` session data? Is this to store it?

File: lib/user.rb [hunk 3]
```diff
@@ -26,4 +26,14 @@ def self.create(username, email, password)
     )
     User.new(result[0]['id'], result[0]['username'], result[0]['email'], result[0]['password'])
   end
+
+  def self.find(id)
+    result = DatabaseConnection.query("SELECT * FROM users WHERE id = $1", [id])
```
Comment (michelle-watkins) [inline lib/user.rb:31]: Use single quotes?? "SELECT * FROM users WHERE id = $1" --> 'SELECT * FROM users WHERE id = $1'

File: app.rb (hunk 2 above)
Comment (lewiscj97) [reply app.rb]: Sorry I just deleted these - they were for testing and I forgot to remove them

File: lib/user.rb (hunk 3 above)
Comment (lewiscj97) [reply lib/user.rb]: addressed

**Pull request merged**